packages/sdk/client/src/packlets/client/client.ts

```diff
@@ -6,35 +6,35 @@ import assert from 'node:assert';
 import { inspect } from 'node:util';
 
 import { Event, MulticastObservable, synchronized, Trigger } from '@dxos/async';
+import {
+  AuthenticatingInvitationObservable,
+  ClientServicesProvider,
+  createDefaultModelFactory,
+  Echo,
+  Space,
+} from '@dxos/client-protocol';
 import type { Stream } from '@dxos/codec-protobuf';
 import { Config } from '@dxos/config';
 import { inspectObject } from '@dxos/debug';
-import { DocumentModel } from '@dxos/document-model';
 import { DatabaseRouter, EchoSchema } from '@dxos/echo-schema';
 import { ApiError } from '@dxos/errors';
 import { PublicKey } from '@dxos/keys';
 import { log } from '@dxos/log';
 import { ModelFactory } from '@dxos/model-factory';
 import { trace } from '@dxos/protocols';
 import { Invitation, SystemStatus, SystemStatusResponse } from '@dxos/protocols/proto/dxos/client/services';
-import { TextModel } from '@dxos/text-model';
+import { isNode } from '@dxos/util';
 
 import { DXOS_VERSION } from '../../version';
 import { createDevtoolsRpcServer } from '../devtools';
-import { AuthenticatingInvitationObservable } from '../invitations';
 import { PropertiesProps } from '../proto';
-import { EchoProxy, HaloProxy, MeshProxy, Space } from '../proxies';
+import { EchoProxy, HaloProxy, MeshProxy } from '../proxies';
 import { SpaceSerializer } from './serializer';
-import { ClientServicesProvider } from './service-definitions';
-import { fromIFrame } from './utils';
+import { fromHost, fromIFrame } from './utils';
 
-// TODO(burdon): Define package-specific errors.
+const defaultServiceProvider = isNode() ? fromHost : fromIFrame;
```

Comment:
since this is only used in one place i think better to move inline where it's set.